src/AstarteManager.tsx

```diff
@@ -99,7 +99,7 @@ type AstarteContextValue = {
   logout: () => void;
 };
 
-const AstarteContext: React.Context<AstarteContextValue> = createContext(null) as any;
+const AstarteContext = (createContext(null) as unknown) as React.Context<AstarteContextValue>;
```

Comment:
can we safelly cast a `unknown` to a `React.Context`? if that works, couldn't we just simplify this as
`createContext(null) as React.Context<AstarteContextValue>`?

Reply:
Typescript cannot cast `createContext(null) as React.Context<AstarteContextValue>`, since it is reading the defaultValue `null` and it infers a type of `AstarteContextValue | null`.
Casting to `unknown` first is therefore an escape hatch, so that components receiving the context value don't have to check for `null` value each time. I'd say we can safely do this because we know that Provider will always have a value and consumer components are always rendered within the scope of the Provider.

Reply:
Typescript is actually right here. The contex may be null! of course we don't want to continuing checking for that in our app.
Check out the second reply on this topic : https://stackoverflow.com/questions/63080452/react-createcontextnull-not-allowed-with-typescript

Reply:
That the context may be null is a correct assumption for the Typescript compiler, however we know we are always passing a value. And you can read from [React docs](https://it.reactjs.org/docs/context.html#reactcreatecontext) that `null`, our defaultValue, is only used when a component does not have a matching Provider above it in the tree

Reply:
However I reworked the hook that consume the context adding a null check, as in the StackOverflow example, so we don't have to resort to escape hatches with Typescript